### PR TITLE
Pull request for lesson 1

### PR DESCRIPTION
**Pull request by nguyenvanman**

### CODE REVIEW

File: ConsoleApp1/Program.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+using System.Threading.Tasks;
+
+namespace ConsoleApp1
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            int num = Convert.ToInt32(Console.ReadLine());
+            if (num == 1) Console.WriteLine("Sunday");
```
Comment (voquanghoa) [inline ConsoleApp1/Program.cs]: `number` is a better name

File: ConsoleApp1/Program.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            else if (num == 2) Console.WriteLine("Monday");
+            else if (num == 3) Console.WriteLine("Tuesday");
+            else if (num == 4) Console.WriteLine("Wednesday");
+            else if (num == 5) Console.WriteLine("Thursday");
+            else if (num == 6) Console.WriteLine("Friday");
+            else if (num == 7) Console.WriteLine("Saturday");
```
Comment (voquanghoa) [inline ConsoleApp1/Program.cs]: Must extract this code to a method like `WriteDateNameWithIf`

File: ConsoleApp1/Program.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            else
+            {
+                Console.WriteLine("Invalid day");
+            }
+
+            switch (num)
```
Comment (voquanghoa) [inline ConsoleApp1/Program.cs]: Extract this to make a method like `WriteDateNameWithSwitchCase`

File: ConsoleApp1.sln [hunk 4]
```diff
@@ -0,0 +1,25 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio 15
+VisualStudioVersion = 15.0.27004.2002
+MinimumVisualStudioVersion = 10.0.40219.1
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ConsoleApp1", "ConsoleApp1\ConsoleApp1.csproj", "{B929E999-8E3A-4987-AF8D-EC2B243F9C62}"
```
Comment (voquanghoa) [inline ConsoleApp1.sln]: For next time, the name of the project should be `Exercise xxx`

**Comment by nguyenvanman** [comment]
Done

File: ConsoleApp1/Program.cs [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+using System.Threading.Tasks;
+
+namespace ConsoleApp1
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            int num = Convert.ToInt32(Console.ReadLine());
+            Console.WriteLine(num);
```
Comment (stevelee2497) [inline ConsoleApp1/Program.cs]: óc chó

File: ConsoleApp1/Program.cs (hunk 1 above)
Comment (nguyenvanman) [reply ConsoleApp1/Program.cs]: Done

File: ConsoleApp1/Program.cs (hunk 5 above)
Comment (nguyenvanman) [reply ConsoleApp1/Program.cs]: uh

File: ConsoleApp1/Program.cs [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+namespace ConsoleApp1
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.Write("Enter a number: ");
+            int number = Convert.ToInt32(Console.ReadLine());
+            //1. if_else
+            if (number == 1)
```
Comment (voquanghoa) [inline ConsoleApp1/Program.cs]: Extract code to a function

File: ConsoleApp1/Program.cs [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+                    break;
+                case 7:
+                    Console.WriteLine("Saturday");
+                    break;
+                default:
+                    Console.WriteLine("Invalid day");
+                    break;
+            }
+            //3.Array
+            String[] listDays = { "Invalid day", "Sunday", "Monday", "Tuesday", "Wedndesday", "Thursday", "Friday", "Saturday" };
```
Comment (voquanghoa) [inline ConsoleApp1/Program.cs]: Extract to a function

File: Exercise5/Program.cs [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+            sort(listNumbers, size);
+
+            for (int i = 0; i < size; i++)
+            {
+                Console.Write(listNumbers[i] + " ");
+            }
+            Console.ReadKey();
+
+        }
+        static void sort(int[] listNumbers, int size)
```
Comment (voquanghoa) [inline Exercise5/Program.cs]: You don't need to add `size` here, just need to use `for (int i = 0; i <listNumbers.Length  - 1; i++)`

Comment (voquanghoa) [inline Exercise5/Program.cs]: You don't need to add size here, just need to use for (int i = 0; i <listNumbers.Length - 1; i++)

File: Exercise6/Program.cs [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+using System.Threading.Tasks;
+
+namespace Exercise6
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.Write("Enter the first edge: ");
+            var firstEdge = Convert.ToInt32(Console.ReadLine());
```
Comment (voquanghoa) [inline Exercise6/Program.cs]: Create a function and call `var firstEdge = ReadInputInt("Enter the first edge: ");`

File: Exercise6/Program.cs [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+            Console.Write("Enter the second edge: ");
+            var secondEdge = Convert.ToInt32(Console.ReadLine());
+            Console.Write("Enter the third edge: ");
+            var thirdEdge = Convert.ToInt32(Console.ReadLine());
+            Console.WriteLine("Type: " + GetTypeOfTriangle(firstEdge, secondEdge, thirdEdge));
```
Comment (voquanghoa) [inline Exercise6/Program.cs]: You should say 
```C#
if(IsEdgesOfTriangle(....))
{
    Console.WriteLine("Type: " + GetTypeOfTriangle(firstEdge, secondEdge, thirdEdge));
    Console.WriteLine("Perimeter: " + CalculatePerimeter(firstEdge, secondEdge, thirdEdge));
    Console.WriteLine("Area: " + CalculateArea(firstEdge, secondEdge, thirdEdge));
}
else
{
    Console.WriteLine(...);
}
```

And remove call `IsEdgesOfTriangle` from `CalculatePerimeter` `GetTypeOfTriangle`

File: Exercise6/Program.cs [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+            }
+            if (firstEdge == secondEdge && secondEdge == thirdEdge)
+            {
+                return "Equilateral Triangle";
+            }
+            if (firstEdge == secondEdge || secondEdge == thirdEdge || thirdEdge == firstEdge)
+            {
+                return "Isosceles Triangle";
+            }
+            var maxEdge = (firstEdge > secondEdge && firstEdge > thirdEdge)
```
Comment (voquanghoa) [inline Exercise6/Program.cs]: it's easier to read if you write
```C#
var sqrFirst = firstEdge*firstEdge;
var sqrSecond = ...
var sqrThird = ...

if(
    (sqrFirst+sqrSecond-sqrThird) *
    (sqrFirst-sqrSecond+sqrThird) *
    (-sqrFirst+sqrSecond + sqrThird) == 0) return "Square Triangle";
```

File: Exercise7/Program.cs [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+            var number = Convert.ToInt32(Console.ReadLine());
+            Console.WriteLine(CalculateFibonaci(number));
+            Console.ReadKey();
+        }
+        static int CalculateFibonaci(int n)
+        {
+            int[] fibonaci = new int[n + 1];
+            fibonaci[0] = 1;
+            fibonaci[1] = 1;
+            for (int i = 2; i <= n; i++)
```
Comment (voquanghoa) [inline Exercise7/Program.cs]: ` for (int i = 2; i <= n; i++)` ==> `for (int i = 2; i < n; i++)`

File: Exercise8/Program.cs [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+using System.Threading.Tasks;
+
+namespace Exercise8
+{
+    class Program
+    {
+        static bool IsPrimeNumber(int number)
+        {
+            if (number < 2) return false;
+            for (int i = 2; i <= Math.Sqrt(number); i++)
```
Comment (voquanghoa) [inline Exercise8/Program.cs]: For better performance, you should write `for (int i = Math.Sqrt(number); i >=2 ; i--)`

File: ConsoleApp1/Program.cs (hunk 6 above)
Comment (nguyenvanman) [reply ConsoleApp1/Program.cs]: Done

File: ConsoleApp1/Program.cs (hunk 7 above)
Comment (nguyenvanman) [reply ConsoleApp1/Program.cs]: Done

File: Exercise6/Program.cs (hunk 9 above)
Comment (nguyenvanman) [reply Exercise6/Program.cs]: Done

File: Exercise5/Program.cs (hunk 8 above)
Comment (nguyenvanman) [reply Exercise5/Program.cs]: Done

File: Exercise6/Program.cs (hunk 10 above)
Comment (nguyenvanman) [reply Exercise6/Program.cs]: Done

File: Exercise6/Program.cs (hunk 11 above)
Comment (nguyenvanman) [reply Exercise6/Program.cs]: Done

File: Exercise7/Program.cs (hunk 12 above)
Comment (nguyenvanman) [reply Exercise7/Program.cs]: Done

File: Exercise8/Program.cs (hunk 13 above)
Comment (nguyenvanman) [reply Exercise8/Program.cs]: Done

**Pull request merged**